src/serializer/ResidualHeapSerializer.js

```diff
@@ -1163,17 +1173,52 @@ export class ResidualHeapSerializer {
     // TODO #20: add timers
 
     // TODO #21: add event listeners
+
     for (let [moduleId, moduleValue] of this.modules.initializedModules)
       this.requireReturns.set(moduleId, this.serializeValue(moduleValue));
 
+    // Make sure additional functions get serialized.
+    // TODO: make sure those functions get skipped over in all relevant places in serialization/optimization
+    // We have to do this if we're on the first pass so effects get reversed properly
+    let rewrittenAdditionalFunctions: Map<FunctionValue, Array<BabelNodeStatement>> = new Map();
+    let processAdditionalFunctionValues = () => {
+      let additionalFVEffects = this.additionalFunctionValuesAndEffects;
+      if (additionalFVEffects) {
+        for (let [additionalFunctionValue, effects] of additionalFVEffects.entries()) {
+          // Create the new function body
+          let [r, g, ob, pb, co] = effects;
+          let oldGenerator = this.generator;
+          let originalLength = g.length;
+          this.generator = g;
```

Comment:
I don't understand the need to mutate `this.generator` and `this.realm.generator`?